File: backend/src/main/java/com/team05/petmeeting/domain/campaign/dto/CampaignCreateRes.java
package com.team05.petmeeting.domain.campaign.dto;

import com.team05.petmeeting.domain.campaign.entity.Campaign;
import com.team05.petmeeting.domain.campaign.enums.CampaignStatus;

public record CampaignCreateRes(
        Long id,
        String title,
        int targetAmount,
        CampaignStatus status
){
    public static CampaignCreateRes from(Campaign campaign) {
        return new CampaignCreateRes(
                campaign.getId(),
                campaign.getTitle(),
                campaign.getTargetAmount(),
                campaign.getStatus()
        );
    }
}

File: backend/src/main/java/com/team05/petmeeting/domain/campaign/dto/CampaignCreateRes.kt
package com.team05.petmeeting.domain.campaign.dto

import com.team05.petmeeting.domain.campaign.entity.Campaign
import com.team05.petmeeting.domain.campaign.enums.CampaignStatus

data class CampaignCreateRes(
    val id: Long,
    val title: String,
    val targetAmount: Int,
    val status: CampaignStatus
) {
    companion object {
        @JvmStatic
        fun from(campaign: Campaign): CampaignCreateRes {
            return CampaignCreateRes(
                campaign.getId(),
                campaign.getTitle(),
                campaign.getTargetAmount(),
                campaign.getStatus()
            )
        }
    }
}
